### ISSUE TITLE
clippy::unwrap_or_default doesn't recognize `vec![]` as default value since 1.92

### ISSUE THREAD

**Issue by rhysd** [rust-lang/rust-clippy#16432]
### Summary

Until 1.91, `clippy::unwrap_or_default` lint reported `unwrap_or` with `vec![]` could be replaced with `unwrap_or_default`. However 1.92 no longer reports it.

### Lint Name

clippy::unwrap_or_default

### Reproducer

I tried this code:

```rust
fn main() {
    let x = Some(Vec::<i32>::new());
    let _y = x.unwrap_or(vec![]);
}
```

`cargo +1.91 clippy` reported as follows:

```
warning: use of `unwrap_or` to construct default value
  --> src\main.rs:45:16
   |
45 |     let _y = x.unwrap_or(vec![]);
   |                ^^^^^^^^^^^^^^^^^ help: try: `unwrap_or_default()`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.91.0/index.html#unwrap_or_default
   = note: `#[warn(clippy::unwrap_or_default)]` on by default
```

However cargo 1.92 reported nothing.

I could not find any item related to this change in the changelog so I thought it was regression in 1.92 and am reporting this.

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-pc-windows-msvc
release: 1.92.0
LLVM version: 21.1.3
```

**Comment by profetia** [comment]
I think this is intentional, since `.unwrap_or(vec![])` is as readable as `.unwrap_or_default()`

**Comment by samueltardieu** [comment]
> I think this is intentional, since `.unwrap_or(vec![])` is as readable as `.unwrap_or_default()`

Confirmed, we even have a test for that:

```rust
    let with_default_vec_macro = Some(vec![1, 2, 3]);
    with_default_vec_macro.unwrap_or(vec![]);
    // Do not lint because `.unwrap_or_default()` wouldn't be simpler
```

Closing the issue since this is not a bug. @rhysd if you have a need for such a feature (triggering on `vec![]`), please open a feature request issue explaining why you need it. Thanks!
